new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Pod.java

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.assertions.impl;
+
+import java.util.ArrayList;
+import java.util.concurrent.Callable;
+
+import org.awaitility.core.ConditionFactory;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class Pod {
+
+  /**
+   * Check pods are restarted in the same order as in the pods list.
+   * @param pods names of the pods in a list
+   * @param namespace name of the namespace in which to check for pods rolling restart
+   * @return true if pods are restarted in rolling fashion
+   */
+  public static boolean verifyRollingRestartOccurred(ArrayList<String> pods, String namespace) {
+
+    // check the pods list is not empty
+    if (pods.isEmpty()) {
+      logger.severe("The pods list is empty");
+      return false;
+    }
+
+    // reusable condition factory
+    ConditionFactory retry
+        = with().pollInterval(5, SECONDS).atMost(5, MINUTES).await();
+
+    // check pods are terminated and started.
+    for (var pod : pods) {
+      retry
+          .conditionEvaluationListener(condition -> logger.info("Waiting for pod {0} to be "
+          + "terminating in namespace {1} "
+          + "(elapsed time {2}ms, remaining time {3}ms)",
+          pod,
+          namespace,
+          condition.getElapsedTimeInMS(),
+          condition.getRemainingTimeInMS()))
+          .until(assertDoesNotThrow(() -> onlyGivenPodTerminating(pods, pod, namespace),
```

Comment:
You assume the pods will be terminated in a specific order. This needs to match how operator does the termination.

Reply:
It is up to the test provide the list of pods in a order for rolling restart

Reply:
The operator does roll the servers in sorted order.